File: 02-backend/ecom-angular-part/src/main/java/ma/codeben/ecom_spring_angular/dto/Purchase.java
package ma.codeben.ecom_spring_angular.dto;

import lombok.Data;
import ma.codeben.ecom_spring_angular.entity.Address;
import ma.codeben.ecom_spring_angular.entity.Customer;
import ma.codeben.ecom_spring_angular.entity.Order;
import ma.codeben.ecom_spring_angular.entity.OrderItem;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: 03-frontend/ecom-angular/src/app/common/purchase.ts
import { OrderItem } from './order-item';
import { Address } from './address';
import { Customer } from './customer';
import { Order } from './order';
export class Purchase {
    custormer!: Customer;
    shippingAddress!: Address;
    billingAddress!: Address;
    order!: Order;
    OrderItems!: OrderItem[];
}
